Dockerfile-dev

```diff
@@ -0,0 +1,50 @@
+# This Dockerfile has three stages:
+#
+# base-image
+#   Updates the base Python image with security patches and common system
+#   packages. This image becomes the base of all other images.
+# dependencies-image
+#   Installs third-party dependencies (requirements/main.txt) into a virtual
+#   environment. This virtual environment is ideal for copying across build
+#   stages.
+# development-image
+#   Installs the app into the virtual environment.
+#   Copies the virtual environment into place.
+#   Sets up an entrypoint.
+
+FROM python:3.7-buster AS base-image
+
+# Update system packages
+COPY scripts/install-base-packages.sh .
+RUN ./install-base-packages.sh
+
+FROM base-image AS dependencies-image
+
+# Create a Python virtual environment
+ENV VIRTUAL_ENV=/opt/venv
+RUN python -m venv $VIRTUAL_ENV
+# Make sure we use the virtualenv
+ENV PATH="$VIRTUAL_ENV/bin:$PATH"
+# Put the latest pip and setuptools in the virtualenv
+RUN pip install --upgrade --no-cache-dir pip setuptools wheel
+
+# Install the app's Python development dependencies, this is required for
+# running the app in development mode with docker-compose
+COPY requirements/dev.txt ./dev.txt
```

Comment:
That's related to my desire to run the app using `docker-compose` during development. One of the reasons for that is to avoid installing RocksDB locally, which would require brew to install. If I could just use an image for development that included RocksDB it would make that easier. But yes, I'll re-evaluate the need for that when I get to use RocksDB.